url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/FormatData2Norm.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * 
+ */
+package utility;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Scanner;
+
+import utility.geo.DataOfLine;
+
+/**
+ * @author chenqian
+ *
+ */
+public class FormatData2Norm {
+
+	public static int scale = 10000;
+	public static ArrayList<long[]> pointsAfter = null;
+	public static HashSet<Long> cnt = new HashSet<Long>();
+	
+	public static long getId(long id1, long id2){
+		return id1 * DataOfLine.M + id2;
+	}
+
+	public static void loadFile(String sourceFileName, String destFileName) throws FileNotFoundException{
+		Scanner filein = new Scanner(new File(sourceFileName));
+		PrintWriter pw = new PrintWriter(new File(destFileName));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;destFileName&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;sourceFileName&#39;. The same parameter type is validated here for example: njr_failed/url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/quickhull3d/InternalErrorException.java:9. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).